File: java/src/test/java/FacadeTest.java
import facade.Bank;
import facade.account.CheckingAccount;
import facade.log.ConsoleLog;
import facade.notification.Notification;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

public class FacadeTest {

    @Test
    void testFacade() {
        final var bank = new Bank(
                new CheckingAccount(10_000_000),
                new ConsoleLog(),
                new Notification()
        );

        bank.depositWithLogAndNotification(5000);
        assertEquals(10_005_000, bank.getBalance());

        bank.withdrawWithLogAndNotification(5000);
        assertEquals(10_000_000, bank.getBalance());

        assertThrows(IllegalArgumentException.class, () ->
                bank.depositWithLogAndNotification(5_000_001)
        );

        assertThrows(IllegalArgumentException.class, () ->
                bank.withdrawWithLogAndNotification(bank.getBalance() + 1)
        );
    }
}

File: src/test/kotlin/FacadeTest.kt
import facade.Bank
import facade.account.CheckingAccount
import facade.log.ConsoleLog
import facade.notification.Notification
import io.kotest.assertions.throwables.shouldThrow
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class FacadeTest : FunSpec({
    test("Test Facade") {
        val bank = Bank(
            CheckingAccount(10_000_000),
            ConsoleLog(),
            Notification()
        )

        bank.depositWithLogAndNotification(5000)
        bank.getBalance() shouldBe 10_005_000

        bank.withdrawWithLogAndNotification(5000)
        bank.getBalance() shouldBe 10_000_000

        shouldThrow<IllegalArgumentException> {
            bank.depositWithLogAndNotification(5000001)
        }

        shouldThrow<IllegalArgumentException> {
            bank.withdrawWithLogAndNotification(bank.getBalance() + 1)
        }
    }
})